File: Backend/participant-service/src/main/java/com/ensa/projet/participantservice/entities/TestResult.java
package com.ensa.projet.participantservice.entities;

import com.ensa.projet.participantservice.dto.TrainingDTO;
import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "test_results")
@Data @Builder @NoArgsConstructor @AllArgsConstructor
public class TestResult {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "participant_id")
    private Integer participantId;

    private Integer trainingId;

    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "test_result_id")
    @Builder.Default
    private List<ParticipantAnswer> userAnswers = new ArrayList<>();

    private float score;
    private LocalDateTime submissionDate;
    private boolean passed;

    @Transient
    private TrainingDTO trainingDTO;
    public void setTrainingDTO(TrainingDTO trainingDTO) {
        this.trainingDTO = trainingDTO;
    }

    public Integer getParticipantId() {
        return participantId;
    }
}

File: Mobile/app/src/main/java/com/example/firstaidfront/models/TestResult.kt
package com.example.firstaidfront.models



data class TestResult(
    val id:Int,
    val participantId: Int,
    val trainingId: Int,
    val userAnswers: List<ParticipantAnswer>,
    val score: Float,
    val submissionDate: String,
    val passed: Boolean,
    val trainingDTO: TrainingDTO
) {
    data class TrainingDTO(
        val id: Int,
        val title: String,
        val description: String,
        val instructions: List<String>,
        val quizzes: List<Any>? = null,
        val estimatedDurationMinutes :Int
    )
}
